### PR TITLE
Add gas scaling relation plots

### PR DESCRIPTION
**Pull request by MatthieuSchaller**

### CODE REVIEW

File: eagle-xl/registration.py [hunk 1]
```diff
@@ -123,15 +123,15 @@
...
+# Now HI masses
 
 try:
     gas_mass = catalogue.masses.m_gas
     H_frac = getattr(catalogue.element_mass_fractions, "element_0")
     HI_frac = getattr(catalogue.species_fractions, "species_0")
 
     HI_mass = gas_mass * H_frac * HI_frac
-    HI_mass.name = "$M_{\\rm HI}$"
+    HI_mass.name = "$M_{HI}$"
```
Comment (JBorrow) [inline eagle-xl/registration.py:134]: Why did we remove the `\\rm`?

File: eagle-xl/registration.py [hunk 2]
```diff
@@ -145,15 +145,15 @@
...
+# Now H2 masses
 
 try:
     gas_mass = catalogue.masses.m_gas
     H_frac = getattr(catalogue.element_mass_fractions, "element_0")
     H2_frac = getattr(catalogue.species_fractions, "species_2")
 
     H2_mass = gas_mass * H_frac * H2_frac
-    H2_mass.name = "$M_{\\rm H_2}$"
+    H2_mass.name = "$M_{H_2}$"
```
Comment (JBorrow) [inline eagle-xl/registration.py:156]: Similar question here

File: eagle-xl/registration.py [hunk 3]
```diff
@@ -166,6 +166,70 @@
...
+    H2_mass = gas_mass * H_frac * H2_frac
+    neutral_H_mass = HI_mass + H2_mass
+    neutral_H_mass.name = "$M_{HI + H_2}$"
+
+    setattr(self, "gas_neutral_H_mass_Msun", neutral_H_mass)
+
+    for aperture_size in aperture_sizes:
+        stellar_mass = getattr(catalogue.apertures, f"mass_star_{aperture_size}_kpc")
+        neutral_H_fraction = neutral_H_mass / stellar_mass
+        neutral_H_fraction.name = "$M_{HI + H_2} / M_*$" + f"({aperture_size} kpc)"
```
Comment (JBorrow) [inline eagle-xl/registration.py:187]:
```suggestion
        neutral_H_fraction.name = f"$M_{{HI + H_2}} / M_*$ ({aperture_size} kpc)"
```

File: eagle-xl/registration.py [hunk 4]
```diff
@@ -166,6 +166,70 @@
...
+
+        molecular_H_fraction = H2_mass / (H2_mass + stellar_mass)
+        molecular_H_fraction.name = (
+            "$M_{H_2} / (M_* + M_{H_2})$" + f"({aperture_size} kpc)"
```
Comment (JBorrow) [inline eagle-xl/registration.py:191]:
```suggestion
            f"$M_{{H_2}} / (M_* + M_{{H_2}})$ ({aperture_size} kpc)"
```

Comment (MatthieuSchaller) [reply eagle-xl/registration.py:191]: The {} do not play well with the f hence that weird construction. And the change to M_HI above to match the style.

**Comment by MatthieuSchaller** [comment]
Done.

**Pull request merged**